src/pages/dashboardGroupEdit/components/GroupForm.tsx

```diff
@@ -293,12 +293,12 @@ export default function GroupForm({
   });
 
   useEffect(() => {
-    if (formik.dirty) {
+    if (formik.dirty || (productSelected && !(initialFormData as PartyGroupOnEdit).productId)) {
```

Comment:
there is already a variable to check if edit
```suggestion
    if (formik.dirty || (productSelected && !isEdit) {
```